workitem/workitem_repository.go

```diff
@@ -110,14 +111,54 @@ func (r *GormWorkItemRepository) Load(ctx context.Context, spaceID uuid.UUID, wi
 	return ConvertWorkItemStorageToModel(wiType, wiStorage)
 }
 
+// LookupIDByNamedSpaceAndNumber returns the work item's ID for the given owner name, space name and item number
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LookupIDByNamedSpaceAndNumber(ctx context.Context, ownerName, spaceName string, wiNumber int) (*uuid.UUID, *uuid.UUID, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "workitem", "LookupIDByNamedSpaceAndNumber"}, time.Now())
+	log.Debug(nil, map[string]interface{}{
+		"wi_number":  wiNumber,
+		"space_name": spaceName,
+		"owner_name": ownerName,
+	}, "Loading work item")
+	query := fmt.Sprintf("select wi.id, wi.space_id from %[1]s wi "+
```

Comment:
Adding a new couple of indexes on `lower(spaces.name)` and `lower(identities.username)` now results in index scans on the 3 tables:
````
postgres=# explain select wi.id from work_items wi
postgres-# join spaces s on wi.space_id = s.id
postgres-# join identities i on s.owner_id = i.id
postgres-# where lower(i.username) = lower('aslak')
postgres-# and wi.number = 199
postgres-# and lower(s.name) = lower('Home')
postgres-# and s.deleted_at is null;
                                  QUERY PLAN
-------------------------------------------------------------------------------
 Nested Loop  (cost=0.00..7.20 rows=1 width=16)
   Join Filter: (s.owner_id = i.id)
   ->  Nested Loop  (cost=0.00..5.06 rows=1 width=32)
         Join Filter: (wi.space_id = s.id)
         ->  Seq Scan on work_items wi  (cost=0.00..3.01 rows=1 width=32)
               Filter: (number = 199)
         ->  Seq Scan on spaces s  (cost=0.00..2.03 rows=1 width=32)
               Filter: ((deleted_at IS NULL) AND (lower(name) = 'home'::text))
   ->  Seq Scan on identities i  (cost=0.00..2.13 rows=1 width=16)
         Filter: (lower(username) = 'aslak'::text)
````